### PR TITLE
Commit lag script with decimal minute precision

### PR DESCRIPTION
**Pull request by lremedi**

### CODE REVIEW

File: troubleshooting/current-commit-lag.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+;with stamps as (
+	select top 1 cast(json_value(stuff(cast(Payload as varchar(max)), 1, 3, ''), '$[0].Body.Timestamp') as datetime) AuditStamp, cast(CommitStamp as datetime) CommitStamp
+	from dbo.Commits
+	order by CheckpointNumber desc
+)
+select
+Latest = AuditStamp,
```
Comment (pmamut) [inline troubleshooting/current-commit-lag.sql:7]: `Occurred`, not `Latest`

File: troubleshooting/current-commit-lag.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+Processed = CommitStamp,
+Behind = CONVERT(FLOAT,CommitStamp-AuditStamp) * 24 * 60, --Decimal Minutes
+Lag = CONVERT(FLOAT,getdate()-CommitStamp) * 24 * 60 --Decimal Minutes
```
Comment (pmamut) [inline troubleshooting/current-commit-lag.sql:10]: `getutcdate()`

File: troubleshooting/current-commit-lag.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+;with stamps as (
+	select top 1 cast(json_value(stuff(cast(Payload as varchar(max)), 1, 3, ''), '$[0].Body.Timestamp') as datetime) AuditStamp, cast(CommitStamp as datetime) CommitStamp
+	from dbo.Commits
+	order by CheckpointNumber desc
+)
+select
+Latest = AuditStamp,
+Processed = CommitStamp,
+Behind = CONVERT(FLOAT,CommitStamp-AuditStamp) * 24 * 60, --Decimal Minutes
```
Comment (pmamut) [inline troubleshooting/current-commit-lag.sql:9]: `CycleTime`, not `Behind` - this is how long it took to process an Audit to produce this Commit

File: troubleshooting/current-commit-lag.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+;with stamps as (
+	select top 1 cast(json_value(stuff(cast(Payload as varchar(max)), 1, 3, ''), '$[0].Body.Timestamp') as datetime) AuditStamp, cast(CommitStamp as datetime) CommitStamp
+	from dbo.Commits
+	order by CheckpointNumber desc
+)
```
Comment (pmamut) [inline troubleshooting/current-commit-lag.sql:5]:
```
,
A as (select top 1 * from dbo.Audit order by ID desc)
```

File: troubleshooting/current-commit-lag.sql (hunk 3 above)
Comment (pmamut) [reply troubleshooting/current-commit-lag.sql:9]: also want `Behind = A.ChangeDateUTC - AuditStamp`

**Review by pmamut — CHANGES_REQUESTED** [review]
not quite

**Comment by pmamut** [comment]
please also check https://github.com/versionone/scripts/blob/master/troubleshooting/current-activity-lag.sql#L6 sort - should probably also be `CommitStamp` or `CheckpointNumber`

**Comment by pmamut** [comment]
and https://github.com/versionone/scripts/blob/master/troubleshooting/current-activity-lag.sql#L7 probably needs to specify `BucketId` and `StreamId` https://github.com/versionone/Core/blob/23.2/VersionOne/EventStore/MetaStream/MetaStreamService.cs#L16-L18

**Comment by lremedi** [comment]
> and https://github.com/versionone/scripts/blob/master/troubleshooting/current-activity-lag.sql#L7 probably needs to specify `BucketId` and `StreamId` https://github.com/versionone/Core/blob/23.2/VersionOne/EventStore/MetaStream/MetaStreamService.cs#L16-L18

https://github.com/versionone/scripts/pull/34/files#diff-f13e9d69147acb2d590a84968b59ca3216d23caf49ef119fa8f185b642c2ab09R9

**Comment by lremedi** [comment]
Better execution plans for current-activity-lag after adding  `BucketId` and `StreamId` filters and ordering by `CheckpointNumber`

image

**Pull request merged**